### ISSUE TITLE
ML — image tagging pipeline for classifier training

### ISSUE THREAD

**Issue by ssmiljanicc** [ssmiljanicc/wyckoff-ai#5]
## Purpose (REFORMULATED 2026-05-25)

Originally framed as "Vision caption pass" — generic alt-text generation for accessibility. After deeper analysis ([discussion in docs/edukacija/02-koncepti-arhitekture.md §3](../docs/edukacija/02-koncepti-arhitekture.md), [PRD Faza-3](../.claude/PRPs/prds/faza-3-trading-and-ml.prd.md)), this is reframed as **ML training data labeling** — assigning Wyckoff-concept tags to each image for use as classifier training set.

## Why this is different from generic captioning

Three discoveries changed the framing:

1. **100% of current images already have author commentary** in the surrounding body text of the source `.md`. Generic captions would duplicate what authors already wrote.
2. **Most chart images have Wyckoff labels baked into pixels** (BCLX, AR, SOW, PSY, LPSY, MSOS, JAC drawn directly on the chart by the author). Images are self-describing visually.
3. **The valuable metadata is categorical tags**, not descriptive sentences — `[spring, phase-c, accumulation]` is what makes images useful as ML training examples (analogous to "squat" / "deadlift" / "bench" labels in the `ml-fitness-tracker` reference repo).

## Task

Build a labeling pipeline that produces:

```json
{
  "image": "raw/bruce_fraser/images/14413148769361250103881.jpg",
  "source_article": "raw/bruce_fraser/posts/articles-wyckoff-2015-09-just-charts.md",
  "tags": ["accumulation", "spring", "phase-c", "secondary-test"],
  "phase": "C",
  "events_visible": ["SC", "AR", "ST", "spring"],
  "structure": "accumulation",
  "asset_class": "stocks",
  "label_source": "llm-extracted-from-context|manual|fraser-annotation",
  "confidence": 0.85
}
```

The output (`data/annotations/manifest.json`) becomes the training set for the M6 phase classifier ([#27 ML — baseline classifier training](https://github.com/ssmiljanicc/wyckoff-ai/issues/27)).

## Approach

1. **First pass: extract tags from author commentary** (deterministic)
   - For each image, read surrounding ~500 chars of body text
   - Match against Wyckoff vocabulary dictionary (from `/CLAUDE.md` §3 required pages)
   - Auto-tag based on which terms appear in nearby text
   - Mark `label_source: llm-extracted-from-context`

2. **Second pass: LLM refinement (optional, opus)**
   - For images where extraction is ambiguous or yields few tags
   - LLM reads full surrounding context, produces refined tag set
   - Used for ~10–20% of images

3. **Spot-check by human** (manual validation on sample)
   - Random 30 images reviewed by Wyckoff-knowledgeable human
   - Adjust extraction heuristics based on disagreements

## Dependencies

- Faza 1 wiki (#7) must be at least 80% complete — phase/event vocabulary must be stable before tagging
- Raw data complete: #2 ✓, #3 ✓, #4 (book pending)

## What this is NOT

- ❌ Generic image accessibility captions (the original framing — defer if ever needed)
- ❌ Vision-based image analysis (the images are self-described by their pixel-baked labels + body text)
- ❌ Real-time tagging — this is for static training data, generated once

## Acceptance

- ≥ 500 tagged images in `data/annotations/manifest.json`
- Each major phase (A/B/C/D/E) has ≥ 75 examples
- Sample of 30 reviewed manually shows ≥ 90% agreement with extracted tags
- Output schema documented in `data/annotations/schema.json`

## Original task content (deferred)

The original framing of "Vision caption pass with Anthropic API for accessibility" is deferred indefinitely. If we ever discover a use case for purely descriptive captions (e.g., orphan images without surrounding context), the technique is straightforward and can be added then.

## Related issues

- [#27 ML — baseline classifier training](https://github.com/ssmiljanicc/wyckoff-ai/issues/27) — consumes this labeled data
- [#25 ML — historical scenario annotation pipeline](https://github.com/ssmiljanicc/wyckoff-ai/issues/25) — sister labeling task for OHLCV windows (not images)
